src/algo/dominators.rs

```diff
@@ -76,6 +79,52 @@ impl<N> Dominators<N>
             None
         }
     }
+
+    /// Returns the dominance frontier of every node in `G`.
+    ///
+    /// From Cytron et. al, the dominance frontier is defined as:
+    /// > the set of all cfg nodes, y, such that b dominates a predecessor of y but does not strictly
+    /// > dominate y
+    ///
+    /// We use the algorithm detailed here: https://www.cs.rice.edu/~keith/EMBED/dom.pdf
```

Comment:
This doc comment should mention big-O runtime complexity.

Reply:
It should also mention that `graph` must be the same, un-mutated graph that this `Dominators` instance was constructed from.

Reply:
Some more context on *why* one might be interested in dominance frontiers would also be nice, ie because they represent join points in a CFG.